### ISSUE TITLE
ArrayBracketSpacingSniff doesn't catch `[ 1, 2, 3 ]`

### ISSUE THREAD

**Issue by caugner** [squizlabs/PHP_CodeSniffer#1727]
My expectation is that`ArrayBracketSpacing` enforces that *that there are no spaces around square brackets.*

However, given the following file *array.php*:

```php
<?php
$array = [ 1, 2, 3 ];

```

The following command does *not* fail (for me):
`phpcs array.php --standard=Squiz --sniffs=Squiz.Arrays.ArrayBracketSpacing`

Why is that?

**Comment by caugner** [comment]
Okay, the cause is that the `ArrayBracketSpacingSniff` only registers for [`T_OPEN_SQUARE_BRACKET` and `T_CLOSE_SQUARE_BRACKET`](https://github.com/squizlabs/PHP_CodeSniffer/blob/d667e245d5dcd4d7bf80f26f2c947d476b66213e/src/Standards/Squiz/Sniffs/Arrays/ArrayBracketSpacingSniff.php#L28-L29), whereas the brackets in `[ 1, 2, 3 ]` are of type `T_OPEN_SHORT_ARRAY` and `T_CLOSE_SHORT_ARRAY`.

**Comment by gsherwood** [comment]
This is because that sniff is not supposed to be checking array definitions - only index array usage. Array definitions have their own rules and enforcing no space here would cause a conflict within the Squiz standard.

**Comment by gsherwood** [comment]
Closing due to lack of additional interest in a new sniff for this
